bot/exts/utilities/bookmark.py

```diff
@@ -123,15 +178,45 @@ async def _bookmark_context_menu_callback(self, interaction: discord.Interaction
     @commands.guild_only()
     @whitelist_override(roles=(Roles.everyone,))
     @commands.cooldown(1, 30, commands.BucketType.channel)
-    async def bookmark(self, ctx: commands.Context) -> None:
-        """Teach the invoker how to use the new context-menu based command for a smooth migration."""
-        await ctx.send(
-            embed=self.build_error_embed(
-                "The bookmark text command has been replaced with a context menu command!\n\n"
-                "To bookmark a message simply right-click (press and hold on mobile) "
-                "on a message, open the 'Apps' menu, and click 'Bookmark'."
+    async def bookmark(
+        self,
+        ctx: commands.Context,
+        target_message: Optional[WrappedMessageConverter],
+        *,
+        title: str = "Bookmark",
+    ) -> None:
+        """
+        Send the author a link to the specified message via DMs.
+
+        Members can either give a message as an argument, or reply to a message.
+
+        Bookmarks can subsequently be deleted by using the `bookmark delete` command in DMs.
+        """
+        target_message: Optional[discord.Message] = target_message or getattr(ctx.message.reference, "resolved", None)
+        if target_message is None:
+            raise commands.UserInputError(MESSAGE_NOT_FOUND_ERROR)
+
+        permissions = ctx.channel.permissions_for(ctx.author)
+        if not permissions.read_messages:
+            log.info(f"{ctx.author} tried to bookmark a message in #{ctx.channel} but has no permissions.")
+            embed = Bookmark.build_error_embed("You don't have permission to view this channel.")
+            await ctx.send(embed=embed)
+            return
+
+        try:
+            await dm_bookmark(ctx.author, target_message, title)
+        except discord.Forbidden:
+            error_embed = self.build_error_embed(
+                    f"{ctx.author.mention}, please enable your DMs to receive the bookmark."
             )
-        )
+            await ctx.send(embed=error_embed)
+        else:
+            log.info(f"{ctx.author.mention} bookmarked {target_message.jump_url} with title '{title}'")
+
+        view = SendBookmark(dm_bookmark, ctx.author, ctx.channel, target_message, title)
```

Comment:
If we create this view before try/except we can add `view.clicked = []` in the except block, to remove the issue Bella mentioned above.

Reply:
Would you mind elaborating on what you mean? I'm not sure I'm following. Thanks!

Reply:
I think what Chris means is that we suppose a user has their DMs open, so you instantiate the `SendBookmark` view before you dm that user. Now the instances has self.clicked assigned with that user's id.
If `dm_bookmark` throws, you'll empty that list (`view.clicked = []`) since the user did click, but didn't receive the bookmark.
This allows them to enable their DMs & click again, which will allow them to receive the bookmark now.